src/components/Layout.tsx

```diff
@@ -5,41 +5,86 @@ import {
+import { type PropsWithChildren } from "react";
 import { useLocation, useNavigate } from "react-router-dom";
+import { SignedIn, SignedOut, SignInButton, SignUpButton, UserButton, useUser } from '@clerk/clerk-react';
+import publicRoutes from "../routes/publicRoutes";
+import protectedRoutes from "../routes/protectedRoutes";
+import {
+  appBarStyles,
+  toolbarStyles,
+  titleStyles,
+  tabsStyles,
+  authBoxStyles,
+  userAvatarStyles,
+  containerStyles
+} from "../styles/Layout.styles";
 
 export default function Layout({ children }: PropsWithChildren) {
   const location = useLocation();
   const navigate = useNavigate();
-  const [tab, setTab] = useState(0);
+  const { isSignedIn, user } = useUser();
+  // const { getToken } = useAuth();
+  console.log(user);
+
+  // const saveToken = getToken().then(data=> console.log(data));
 
-  useEffect(() => {
-    // Update the selected tab based on current path
-    if (location.pathname === "/") setTab(0);
-    else if (location.pathname === "/plot") setTab(1);
-  }, [location.pathname]);
+  // console.log(saveToken);
 
-  const handleChange = (_: React.SyntheticEvent, newValue: number) => {
-    setTab(newValue);
-    if (newValue === 0) navigate("/");
-    else if (newValue === 1) navigate("/plot");
+  const allTabs = isSignedIn
+    ? [...publicRoutes, ...protectedRoutes]
+    : publicRoutes;
+
+  const handleChange = (_: React.SyntheticEvent, newValue: string) => {
+    navigate(newValue); // newValue is now pathname
   };
 
   return (
     <>
-      <AppBar position="static">
-        <Toolbar>
-          <Typography variant="h6" sx={{ flexGrow: 1 }}>
+      <AppBar sx={appBarStyles}>
+        <Toolbar sx={toolbarStyles}>
+          <Typography variant="h6" sx={titleStyles}>
             Turbo Trade - Backtest Dashboard
           </Typography>
-          <Tabs value={tab} onChange={handleChange} textColor="inherit" indicatorColor="secondary">
-            <Tab label="Home" />
-            <Tab label="Plot" />
+          <Tabs
+            value={location.pathname}
+            onChange={handleChange}
+            textColor="inherit"
+            indicatorColor="secondary"
+            sx={tabsStyles}
+          >
+            {allTabs.map((tab, index) => (
+              <Tab key={index} label={tab.label} value={tab.path} />
+            ))}
           </Tabs>
+          <Box sx={authBoxStyles}>
+            <SignedOut>
+              <SignInButton mode="modal">
+                <Button sx={{ color: 'white'}} >
+                  Sign In
+                </Button>
+              </SignInButton>
+              <SignUpButton mode="modal" >
+                <Button sx={{ color: 'white' }} >
+                  Sign Up
+                </Button>
+              </SignUpButton>
+            </SignedOut>
+
+            <SignedIn>
+              <UserButton
+                appearance={{
+                  elements: { userButtonAvatarBox: userAvatarStyles },
```

Comment:
Move this part in styles as well. Appearance prop completely can come from styles.